protocol/contracts/zaps/TempleZaps.sol

```diff
@@ -0,0 +1,504 @@
+   * @param _minStableReceived Minimum temple pair stable token to receive
+   * @param _swapTarget Execution target for the swap
+   * @param _swapData DEX data
+   */
+  function zapInTemple(
+    address _fromToken,
+    uint256 _fromAmount,
+    uint256 _minTempleReceived,
+    address _stableToken,
+    uint256 _minStableReceived,
+    address _swapTarget,
+    bytes memory _swapData
+  ) external payable whenNotPaused {
+    zapInTempleFor(_fromToken, _fromAmount, _minTempleReceived, _stableToken, _minStableReceived, msg.sender, _swapTarget, _swapData);
+  }
+
+  /**
+   * @notice This function zaps ETH or an ERC20 token to TEMPLE LP token
+   * @param _fromAddress The token used for entry (address(0) if ether)
+   * @param _fromAmount The amount of fromToken to zap
+   * @param _minAmountOut Minimum tokens out after first DEX swap
+   * @param _swapTarget Execution target for the swap
+   * @param _params Parameters for liquidity addition
+   * @param _swapData DEX data
+   */
+  function zapInTempleLP(
+    address _fromAddress,
+    uint256 _fromAmount,
+    uint256 _minAmountOut,
+    address _swapTarget,
+    TempleLiquidityParams memory _params,
+    bytes memory _swapData
+  ) external payable whenNotPaused {
+    zapInTempleLPFor(_fromAddress, _fromAmount, _minAmountOut, msg.sender, _swapTarget, _params, _swapData);
+  }
+
+  /**
+   * @notice This function zaps ETH or an ERC20 token to TEMPLE and stakes in core vault
+   * @param _fromToken The token used for entry (address(0) if ether)
+   * @param _fromAmount The amount of fromToken to zap
+   * @param _minTempleReceived Minimum tokens out after first DEX swap
+   * @param _stableToken Supported temple pair stable token
+   * @param _minStableReceived Minimum stable token to receive
+   * @param _vault Target core vault
+   * @param _swapTarget Execution target for the swap
+   * @param _swapData DEX data
+   */
+  function zapInVault(
+    address _fromToken,
+    uint256 _fromAmount,
+    uint256 _minTempleReceived,
+    address _stableToken,
+    uint256 _minStableReceived,
+    address _vault,
+    address _swapTarget,
+    bytes memory _swapData
+  ) external payable whenNotPaused {
+    zapInVaultFor(_fromToken, _fromAmount, _minTempleReceived, _stableToken, _minStableReceived, _vault, msg.sender, _swapTarget, _swapData);
+  }
+
+  /**
+   * @notice This function zaps ETH or an ERC20 token to TEMPLE in addition to all eligible usable faith (for boost) and stakes in core vault
+   * @param _vault Target core vault
+   * @param _fromToken The token used for entry (address(0) if ether)
+   * @param _fromAmount The amount of fromToken to zap
+   * @param _minTempleReceived Minimum tokens out after first DEX swap
+   * @param _stableToken Supported temple pair stable token
+   * @param _minStableReceived Minimum stable token to receive
+   * @param _swapTarget Execution target for the swap
+   * @param _swapData DEX data
+   */
+  function zapTempleFaithInVault(
```

Comment:
We're no longer supporting faith in core (as in, not allowing any more deposits via faith) so we can remove any faith related code from this contract. (sorry man)